File: dp/LC70.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        ls=[]
        ls.append(1)
        ls.append(2)
        if(n==1):return 1
        if(n==2):return 2
        for i in range (3,n+1):
            ls.append(ls[i-1]+ls[i-2])
        return ls[n-1]

File: dp/LC70.java
package dp;

public class LC70 {
    public int climbStairs(int n) {
        int[] list=new int[n];
        list[0]=1;
        if(n==1){
            return 1;
        }
        list[1]=2;
        for(int i=2;i<n;i++){
            list[i]=list[i-1]+list[i-2];
        }
        return list[n-1];
    }
}